topic-operator/src/main/java/io/strimzi/operator/topic/ZkTopicsWatcher.java

```diff
@@ -84,6 +84,7 @@ void start(Zk zk) {
                     LogContext logContext = LogContext.zkWatch(TOPICS_ZNODE, "-" + topicName);
                     topicOperator.onTopicDeleted(logContext, new TopicName(topicName)).setHandler(ar -> {
                         if (ar.succeeded()) {
+                            topicOperator.decrementTopicCounter();
```

Comment:
This looks a bit fragile, but I assume it works. The other operator set the topic counter in the `reconcileAll` method which easily gets the count of all topics with every periodic reconciliation. But not sure that is any easier or harder in the TO.